### ISSUE TITLE
Tilix-quake: always on top

### ISSUE THREAD

**Issue by luispabon** [gnunn1/tilix#1301]
Coming from Terminator, I have some acquired habits. This is one of them: I'd love tilix-quake to have the option of always being on top.

**Comment by gnunn1** [comment]
It's always on top by default and I do not plan to make it optional. Note it doesn't work under Wayland though and there is nothing I can do about it there.

**Comment by luispabon** [comment]
gaah wayland and its limitations

**Comment by guillefar** [comment]
Can I ask why you don't have plans to make "always on top" optional? I think having that option can be really helpful. 
In my case, I use several screens, and it is a lot more comfortable to open the terminal with a key, and then going out of it with by clicking with the mouse in the window in the back of Tilix. And while "Hide window when focus is lost" seems it might be an option in this case, it really isn't, because if I activated I can't do something really basic as having a some document in the other screen, and copying and pasting  from there to the console. 
Is there a technical limitation for it?

**Comment by gnunn1** [comment]
Mostly because it didn't work under Wayland and I didn't want to add options that wouldn't work the same everywhere considering how much the options were already different between X11 and Wayland. Now that I have removed the Wayland support for quake mode I'm open to re-considering this.

**Comment by guillefar** [comment]
Ah, nice to know. Thanks.

**Comment by luispabon** [comment]
You might want to reconsider soon re-adding wayland support for quake mode, Ubuntu will likely revert back to wayland as a default on 18.10 later this year.

**Comment by gnunn1** [comment]
Wayland support is not coming back unless Wayland starts allowing the positioning of windows either directly or via some other method.

**Comment by Guddler** [comment]
+1 for the opposite of what the title says :)

I tried Quake mode a long time ago and quickly dismissed it. I don't recall why but just recently I've really taken to it. I spend all day working on a large JavaEE application. I have JBoss running in the background in a quake based Tilix terminal.

It would be the icing on the cake if I could bring the terminal down with my key shortcut and keep an eye on it in the background behind the browser. Can't do that at the moment because it obscures what I'm trying to do in the browser.

**Comment by guillefar** [comment]
Just got to try this new functionality, it is almost there, but there is an issue. 
Indeed, I can click in another windows and Tilix will disappear (or actually lost focus and go to the back of the window) as expected. However, using F12 again don't bring up Tilix again as expected. At least in KDE Plasma, Tilix will stay in the back of the windows, make it effectively invisible if the window in front is maximized. 
Perhaps a solution will be when F12 is press, to either activate momentarily "Always on top"  or to instruct the Tilix window to grab focus and place itself in the front, which I think is what everyone would expect when calling Tilyx using the F12 button. 
Please let me know if I wasn't clear of if you need I test something in my system. 
Cheers.

**Comment by gnunn1** [comment]
Do you have the new Always on top option checked? All it does is the gtk keep on top option, if plasma is not respecting that by keeping the window on top there is not much I can do.

Let me know if I'm mis-understanding.

**Comment by guillefar** [comment]
Actually the problem is when "Always in top" it is not enabled. I think it is useful for many workflows for the Tilyx window to be actually to lose focus and go to the back to "go out" of it.  However, I want for it to appear and on top of my other windows if I press F12.
As is working right now, if "Always in top" it is not enabled, what happen is that when the Tilix window goes to the back, F12 doesn't bring it to the front, it will not even make it visible if the window in front is maximized. Tilix just open and close but in the back of the other windows. 
Basically, I think F12 should make appear the Tilix window in front always, despite if the Tilix window is marked as "Always on top". Then yes, the window could stay always visible or not depending on the option. 
Hope that's more clear and thanks for the quick reply.

**Comment by jpzbkk** [comment]
@gnunn1 Is there any interest in supporting Quake mode to not always be on top, like as discussed in this thread?  Ideally I want it to gain focus when hitting my hotkey, but lose #1 placement when clicking another app So I can see both.
